File: scripts/aux/spellChecker.pl
#!/usr/bin/env perl
use strict;
use warnings;
use lib $ENV{'GALACTICUS_EXEC_PATH'}."/perl";
use LaTeX::SpellCheck;

# Perform spell checking of files
# Andrew Benson (28-February-2023)

# Get the file to process.
die('Usage: spellChecker.pl <fileName> <warningFile>')
    unless ( scalar(@ARGV) == 2 );
my $fileName        = $ARGV[0];
my $warningFileName = $ARGV[1];

# Check the file.
my $warnings = &LaTeX::SpellCheck::spellCheckFile($fileName,$fileName);

# Write accumulated warnings to file (appending).
unless ( $warnings eq "" ) {
    open(my $warningFile,">>",$warningFileName);
    print $warningFile $warnings;
    close($warningFile);
}

exit 0;

File: scripts/aux/spellChecker.py
#!/usr/bin/env python3
import os
import sys
import latex_spellcheck

# Perform spell checking of files.
# Andrew Benson (28-February-2023) [Python port]

if len(sys.argv) != 3:
    print("Usage: spellChecker.py <fileName> <warningFile>", file=sys.stderr)
    raise SystemExit(1)

file_name         = sys.argv[1]
warning_file_name = sys.argv[2]

# Check the file.
warnings = latex_spellcheck.spell_check_file(file_name, file_name)

# Append any warnings to the warning file.
if warnings:
    with open(warning_file_name, 'a') as fh:
        fh.write(warnings)
